rootfs/etc/nginx/lua/test/configuration_test.lua

```diff
@@ -0,0 +1,131 @@
+local configuration = require("configuration")
+local unmocked_ngx = _G.ngx
+
+function get_mocked_ngx_env()
+    local _ngx = {
+        status = 100,
+        print = function(msg) return msg end,
+        var = {},
+        req = {
+            read_body = function() end,
+            get_body_data = function() end,
+            get_body_file = function() end,
+        }
+    }
+    setmetatable(_ngx, {__index = _G.ngx})
+    return _ngx
+end
+
+describe("Configuration", function()
+    before_each(function()
+        _G.ngx = get_mocked_ngx_env()
+    end)
+
+    after_each(function()
+        _G.ngx = unmocked_ngx
+        package.loaded["configuration"] = nil
+        configuration = require("configuration")
+    end)
+
+    context("Request method is neither GET nor POST", function()
+        it("should log 'Only POST and GET requests are allowed!'", function()
+            ngx.var.request_method = "PUT"
+            local s = spy.on(ngx, "print")
```

Comment:
We would need to modify: `ngx.print("Only POST and GET requests are allowed!")` on `configuration.lua` to instead use `ngx.log`? Otherwise the test fails.

Reply:
I misinterpreted `ngx.print` functionality, I believe I should fix the title